include/dlaf/mc/cholesky.h

```diff
@@ -0,0 +1,309 @@
+                        uplo, blas::Op::NoTrans, -1.0, mat.read(LocalTileIndex{j, k}), 1.0,
+                        std::move(mat(LocalTileIndex{j, j})));
+
+          for (SizeType i = j + 1; i < nrtile; ++i) {
+            // Update remaining trailing matrix mat(i,j), reading mat.read(i,k) and mat.read(j,k), using
+            // gemm (blas operation)
+            hpx::dataflow(trailing_matrix_executor, hpx::util::unwrapping(tile::gemm<T, Device::CPU>),
+                          blas::Op::NoTrans, blas::Op::ConjTrans, -1.0, mat.read(LocalTileIndex{i, k}),
+                          mat.read(LocalTileIndex{j, k}), 1.0, std::move(mat(LocalTileIndex{i, j})));
+          }
+        }
+      }
+    }
+    else {
+      throw std::runtime_error("uplo = Upper not yet implemented");
+    }
+  }
+  else {
+    throw std::runtime_error("Matrix is not on local memory");
+  }
+}
+
+/// @brief Cholesky implementation on distributed memory
+///
+/// @parame grid refers to a comm::CommunicatorGrid object
+/// @param uplo specifies that the matrix is \a Lower triangular
+/// @tparam mat refers to a dlaf::Matrix object
+///
+/// @throws std::runtime_error if \p uplo = \a Upper decomposition is chosen (not yet implemented)
+
+template <class T>
+void cholesky(comm::CommunicatorGrid grid, blas::Uplo uplo, Matrix<T, Device::CPU>& mat) {
+  // Set up executor on the default queue with high priority.
+  hpx::threads::scheduled_executor matrix_HP_executor =
+      hpx::threads::executors::pool_executor("default", hpx::threads::thread_priority_high);
+  // Set up executor on the default queue with default priority.
+  hpx::threads::scheduled_executor matrix_normal_executor =
+      hpx::threads::executors::pool_executor("default", hpx::threads::thread_priority_default);
+
+  auto col_comm_size = grid.colCommunicator().size();
+  auto row_comm_size = grid.rowCommunicator().size();
+
+  // Set up an executor on the mpi pool --> This part need to be added
+  //  hpx::threads::scheduled_executor mpi_executor;
+  //  if (use_pools && m_size > 1) {
+  //    hpx::threads::executors::pool_executor mpi_exec("mpi");
+  //    mpi_executor = mpi_exec;
+  //   }
+  //  else {
+  //    mpi_executor = matrix_HP_executor;
+  //  }
+
+  // Check if matrix is square
+  util_matrix::assert_size_square(mat, "Cholesky", "mat");
+  // Check if block matrix is square
+  util_matrix::assert_blocksize_square(mat, "Cholesky", "mat");
+
+  const dlaf::matrix::Distribution& distr = mat.distribution();
+
+  SizeType nrtile = mat.nrTiles().cols();
+
+  auto localnrtile_rows = distr.localNrTiles().rows();
+  auto localnrtile_cols = distr.localNrTiles().cols();
+
+  dlaf::common::Pipeline<comm::CommunicatorGrid> serial_comm(std::move(grid));
+
+  // Method only for Lower triangular matrix
+  if (uplo == blas::Uplo::Lower) {
+    for (SizeType k = 0; k < nrtile; ++k) {
+      // Create the panel as a vector of shared future
```

Comment:
The panel is not created here. The vector is just a placeholder in which the shared futures representing the panel will be added.